### DISCUSSION TITLE
Is there any reason why `get` and `get_private` `CookieJar` functions have a different return value?

### DISCUSSION THREAD

**Discussion by Luni-4** [rwf2/Rocket#2639] [Questions] [answered]
Looking at the `CookieJar` documentation for `rocket0.5.0-rc.4` I've noticed that `get` retrieves an `Option<&Cookie<'static>>` as returning value, while `get_private` returns an `Option<Cookie<'static>>`. Is there a particular reason for this difference in these APIs? Thanks for your reply!

**Comment by SergioBenitez** [comment, marked as answer]
Yes. `get()` returns a cookie where the internal value borrows from the request, since its value is identical (modulo percent decoding, which we cache) to what the client sent. `get_private()` fetches a private cookie, whose value, on the other hand, is encrypted (and signed). The logical value, the decrypted value, is thus different than the value already in memory, and thus we cannot borrow it from the request. We must allocate to store said value, hence the owned-ness of the returned `Cookie`.

**Reply by Luni-4** [reply]
Oh, I see. Thanks a lot for your reply!
